### ISSUE TITLE
Plotly doesn't works with python-shell

### ISSUE THREAD

**Issue by LRipi** [extrabacon/python-shell#187]
Hi,

I'm actually developping an application and I would like to use plotly.py with python-shell to use the conversion with plotly.js.

I'm actually developping in node v12.4.0 for Back/Front and Python2.7 to use plotly (seems actually better on this version) with Fedora 30, using PyCharm and WebStorm (from Jetbrains) as IDE.

I've already tried to `pip install plotly` before to run the application and to add options to python-shell, but nothing worked.

-----

Here is the input/output of my last try.

Here is my code in Js :
```js
const options = {
      mode: 'text',
      pythonPath: '/usr/bin/python2.7',
      pythonOptions: ['-u'],
      scriptPath: './',
      args: []
    };
    PythonShell.run('./example.py', null, function (err) {
      if (err) throw err;
       console.log('FINISH')
    })
```

And here is my python script : 
```py

import plotly.graph_objs as go
import plotly.io as plio

import numpy as np


# This function is used to insert your values into the plotly graph.
# This function is called in the main to populate data.
def plotly_values():
    y0 = np.random.randn(50) + 1.2
    y1 = np.random.randn(50) + 1
    y2 = np.random.randn(50) * 2
    y3 = np.random.randn(50) * 0.8 + 1

    trace0 = go.Box(
        y=y0,
        name='Sample A'
    )
    trace1 = go.Box(
        y=y1,
        name='Sample B'
    )
    trace2 = go.Box(
        y=y2,
        name='Sample C'
    )
    trace3 = go.Box(
        y=y3,
        name='Sample D'
    )
    data = [trace0, trace1, trace2, trace3]
    return data


# This function is used to modify the title of the graph.
# This function return the layout, and only this variable should be modified.
def plotly_layout():
    layout = "INSERT TITLE HERE"
    return layout


# This function is the main of the project.
# This function call plotly_values to get the data for the plotly graph.
# This part shouldn't be modified.
def main():
    data = plotly_values()
    layout = go.Layout(title=plotly_layout())
    figure = go.Figure(data=data, layout=layout)
    plio.write_json(figure, './example.json')
    exit(0)


if __name__ == "__main__":
    main()
```

And I got this output :

```
Error: ModuleNotFoundError: No module named 'plotly'
    at PythonShell.parseError (/home/hakken/delivery/Publifactory/publifactory-project/node_modules/python-shell/index.js:246:21)
    at terminateIfNeeded (/home/hakken/delivery/Publifactory/publifactory-project/node_modules/python-shell/index.js:129:32)
    at ChildProcess.<anonymous> (/home/hakken/delivery/Publifactory/publifactory-project/node_modules/python-shell/index.js:121:13)
    at ChildProcess.emit (events.js:189:13)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
    ----- Python Traceback -----
    File "example.py", line 2, in <module>
      import plotly.graph_objs as go
```

I probably did a mistake or just didn't specify the repository of the lib, but I didn't see any docs / issues on this thing. 

-----

I will update this issue if I find a solution before your answers.

**Comment by Almenon** [comment]
Are you sure plotty is installed to python 2 and not python 3? Try doing /usr/bin/python2.7 -m pip freeze and see if plotty is there.  Also I would highly discourage the use of python 2, it's being deprecated in 2020

**Comment by LRipi** [comment]
Hi @Almenon ,

Yes plotty is here.
I've done the following command : 

```bash
➜  plotly_test /usr/bin/python3 -m pip freeze | grep plotly 
➜  plotly_test /usr/bin/python2 -m pip freeze | grep plotly
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
plotly==3.10.0
```
I know the support of Python2.7 won't be maintained after that.

Do you have another answer / possible solutions to help me ?

**Comment by Almenon** [comment]
Weird.  And I'd you try running the below in the python 2 console do you get the same error?

import plotly.graph_objs as go
import plotly.io as plio

**Comment by LRipi** [comment]
Ok, I've finally found my problem and it was like... ***black magic***

When we had our discussion, the problem was plotly ; I've juste reboot my computer and running the script again and something changed : now plotly wasn't the problem, but numpy was.

I can't really explain how the dependance was resolve by itself, then I prefered to go forward and fix this dependency.

I've switch to python3, as you recommended it @Almenon , and installed the package needed (numpy and plotly, to be sure both were installed).

*******

Here is my soluce with the code : 

1. Install Plotly and numpy with this following command 
```bash
sudo python3.7 /usr/bin/pip install numpy plotly
```
I prefered to use absolute path, to avoid *mystic* problems as we already had.

2. Changed the options parameter in my nodeJs code :
```js
    const options = {
        mode: 'text',
        pythonPath: '/usr/bin/python3',
        pythonOptions: ['-u'],
        scriptPath: './',
        args: []
    };
```

3. Check installation before running the script:
```bash
/usr/bin/python3.7 -m pip freeze | grep numpy && /usr/bin/python3.7 -m pip freeze | grep plotly
```

*****

Thanks for replying to this gist, I close it but if a problem / question are needed open it again.
